src/Tools/dotnet-monitor/EndpointInfo/ServerEndpointInfoSource.cs

```diff
@@ -123,6 +117,9 @@ public async ValueTask DisposeAsync()
                 {
                 }
             }
+
+            // Clear all endpoints
+            await _endpointTracker.ClearAsync(CancellationToken.None);
```

Comment:
I think this should be the responsibility of the IEndpointTracker implementation, presumably in an IAsyncDisposable.DisposeAsync method implementation rather than this implementation telling it how to manage its endpoint list.